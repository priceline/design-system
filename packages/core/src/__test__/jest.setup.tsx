import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
