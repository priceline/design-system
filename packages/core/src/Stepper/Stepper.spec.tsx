import React from 'react'
import { render } from '../__test__/testing-library'
import { Stepper } from './Stepper'

const children = (
  <React.Fragment>
    <Stepper.Step completed>First Step</Stepper.Step>
    <Stepper.Step active>Second Step</Stepper.Step>
    <Stepper.Step>Third Step</Stepper.Step>
  </React.Fragment>
)

describe('Stepper', () => {
  test('renders without crashing', () => {
    const { asFragment } = render(<Stepper />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with children', () => {
    const { asFragment } = render(<Stepper>{children}</Stepper>)
    expect(asFragment()).toMatchSnapshot()
  })
})
