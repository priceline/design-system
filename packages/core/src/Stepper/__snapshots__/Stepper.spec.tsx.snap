// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper renders with children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  color: #0068ef;
  width: 16px;
  margin-right: 4px;
}

.c6 {
  outline: none;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 9999px;
  border-width: 0;
  border-style: solid;
  border-radius: 9999px;
  font-size: 14px;
  padding: 9.5px 18px;
  background-color: #0068ef;
  color: #fff;
  margin-right: 32px;
}

.c2:disabled {
  cursor: not-allowed;
  color: #879db3;
  background-color: #edf0f3;
}

.c2:hover {
  background-color: #049;
  color: #fff;
}

.c2:focus-visible {
  outline: 0px solid #049;
  box-shadow: 0 0 0 2px #049;
}

.c7 {
  color: #0068ef;
  font-size: 14px;
}

.c8 {
  color: #0068ef;
  font-weight: 700;
  font-size: 14px;
}

.c9 {
  color: #4f6f8f;
  font-size: 14px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  cursor: default;
  padding: 0;
  font-weight: unset;
}

.c3:hover {
  background-color: #fff;
}

.c1 > :not(:last-child) {
  margin-right: 32pxpx;
}

<div
  className="c0 c1"
>
  <button
    className="c2 c3"
    color="primary"
    size="medium"
  >
    <div
      className="c4"
    >
      <svg
        aria-hidden={false}
        aria-label="Step Completed"
        className="c5 c6"
        color="primary"
        fill="currentcolor"
        focusable={false}
        height={16}
        role="img"
        tabIndex={-1}
        viewBox="0 -960 960 960"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M382-354l339-339q12-12 28-12t28 12q12 12 12 28.5T777-636L410-268q-12 12-28 12t-28-12L182-440q-12-12-11.5-28.5T183-497q12-12 28.5-12t28.5 12l142 143z"
        />
      </svg>
      <div
        aria-selected={false}
        className="c7"
        color="primary"
        fontSize={1}
      >
        First Step
      </div>
    </div>
  </button>
  <button
    className="c2 c3"
    color="primary"
    size="medium"
  >
    <div
      className="c4"
    >
      <div
        aria-selected={true}
        className="c8"
        color="primary"
        fontSize={1}
      >
        Second Step
      </div>
    </div>
  </button>
  <button
    className="c2 c3"
    color="primary"
    size="medium"
  >
    <div
      className="c4"
    >
      <div
        aria-selected={false}
        className="c9"
        color="text.light"
        fontSize={1}
      >
        Third Step
      </div>
    </div>
  </button>
</div>
`;

exports[`Stepper renders without crashing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 > :not(:last-child) {
  margin-right: 32pxpx;
}

<div
  className="c0 c1"
/>
`;
